### PR TITLE
Deprecate Radio widget in favor of new RadioGroup widget

### PR DESCRIPTION
**Pull request by toaster**
### Description:

This PR introduces a new `RadioGroup` widget (derived from the old one with `RadioItem` separated).
This will allow proper focus and keyboard control.
The old `Radio` widget is deprecated.

### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

#### Where applicable:

- [x] Public APIs match existing style.
- [x] Any breaking changes have a deprecation path or have been discussed.

### CODE REVIEW

File: cmd/fyne_demo/screens/widget.go [hunk 1]
```diff
@@ -185,9 +185,9 @@ func makeInputTab() fyne.CanvasObject {
 	selectEntry.PlaceHolder = "Type or select"
 	disabledCheck := widget.NewCheck("Disabled check", func(bool) {})
 	disabledCheck.Disable()
-	radio := widget.NewRadio([]string{"Radio Item 1", "Radio Item 2"}, func(s string) { fmt.Println("selected", s) })
+	radio := widget.NewRadioGroup([]string{"RadioGroup Item 1", "RadioGroup Item 2"}, func(s string) { fmt.Println("selected", s) })
```
Comment (andydotxyz) [inline cmd/fyne_demo/screens/widget.go:188]: Should these strings not remain "Radio Item x"?

File: widget/radio_item.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+var _ fyne.Tappable = (*RadioItem)(nil)
+
+func newRadioItem(label string, onTap func(*RadioItem)) *RadioItem {
+	i := &RadioItem{Label: label, onTap: onTap}
+	i.ExtendBaseWidget(i)
+	return i
+}
+
+// RadioItem is a single radio item to be used by RadioGroup.
+type RadioItem struct {
```
Comment (andydotxyz) [inline widget/radio_item.go:22]: I was thinking more about this, I can't think of any benefit to exporting this type?

File: cmd/fyne_demo/screens/widget.go (hunk 1 above)
Comment (toaster) [reply cmd/fyne_demo/screens/widget.go:188]: Oops … yes!

File: widget/radio_item.go (hunk 2 above)
Comment (toaster) [reply widget/radio_item.go:22]: Hm … me neither 😄. Will unexport it.

**Review by andydotxyz — APPROVED** [review]

**Pull request merged**